### ISSUE TITLE
Vendors need ability to manage customer's personal information

### ISSUE THREAD

**Issue by SamPatt** [OpenBazaar/openbazaar-go#834]
Currently a buyer's name, address, and other contact information is permanently stored along with the transactions details, even after a purchase is completed. This is a privacy concern. Vendors should be able to delete completed transactions, or delete the name, address, and contact information at least.

We could include a feature that would automatically delete this information after the transaction has closed. A vendor could decide if they want this feature on or off.

**Comment by SamPatt** [comment]
Related to https://github.com/OpenBazaar/openbazaar-desktop/issues/1734

**Comment by placer14** [comment]
Thinking through this problem, some concerns:

1. Wallet transaction immutability
Data dependencies look a little like: `(node identity)<-(order)<-(wallet tx(s))`
Wallet transactions may be deleted without affecting order integrity. Order AND wallet txs may be deleted together without affecting the node. The problem is the multiwallet uses HD and will redetect those transactions forever and would create orphaned transactions when repopulated without the orders data associated with them.

2. Ricardian Contract signatures: Any obfuscation of the contract data would invalidate the contract.

A solution to this problem will likely require a new "state" of the order called "obfuscated" or "deleted" which comes from all currently final states. This state wouldn't have any data associated with it contract-wise, but the record would be intact to associate with any existing wallet transactions to be handled appropriately by the UI. At least this much would be necessary for any UX designed around "deleted contracts" to work in the UI. There are likely other solutions, but this seems the most obvious and straightforward to me.

**Comment by SamPatt** [comment]
>There are likely other solutions, but this seems the most obvious and straightforward to me.

Indeed, and if I'm understanding this correctly it would likely be usable on existing contracts as well.

There are alternatives, something like placing all personal information within a contract into a specific portion that is encrypted with a different key by the vendor, and then that key being deleted automatically as soon as the order is completed, so the contract remains but the personal information is unreadable. The RC signature would still be valid since the contract hasn't changed.

But that's not backwards compatible and it seems clunky to use separate keys.
